### DISCUSSION TITLE
How to test a service worker?

### DISCUSSION THREAD

**Discussion by dethstrobe** [cypress-io/cypress#15832] [Questions and Help] [answered]
So I'm using a service worker to intercept http requests and append a JWT to the header. I actually got this working which is cool, but I'd like to assert that the Authorization header was added.

Here is what I thought might work
```
      cy.get('@api')
        .its('request.headers')
        .its('Authorization')
        .should('contain', `Bearer ${accessToken}`);
```

However it fails. Because while the service worker successfully intercepted the http request and made a new request with a modified header; the original request never saw it. Which I believe is what cypress is attempting to assert on.

![image](https://user-images.githubusercontent.com/6279187/113787443-e350b700-96ef-11eb-9515-602c995c3177.png)

Anyway, is there a way I can check the service worker request and assert on that?

**Comment by bahmutov** [comment]
Can you provide the reproducible example showing the app and the test?

**Reply by dethstrobe** [reply]
I'll see what i can do

**Reply by dethstrobe** [reply]
I just created a very simple naive example. You can see it here. https://github.com/dethstrobe/sw-cypress

And the thing is, it works as I'd expect it. So I'll go over my more complex app and see if I can't figure out what the difference is. Thanks for taking the time to look at this. I'll try and report back if I find what the problem is.

**Reply by dethstrobe** [reply]
The problem is that the `Authorization` header is `authorization` in cypress. 

So I just needed to change my assert to 
```
      cy.get('@api')
        .its('request.headers')
        .its('authorization')
        .should('contain', `Bearer ${accessToken}`);
```

**Comment by dethstrobe** [comment, marked as answer]
The problem is that the Authorization header is authorization in cypress.

So I just needed to change my assert to

      cy.get('@api')
        .its('request.headers')
        .its('authorization')
        .should('contain', `Bearer ${accessToken}`);
